### PR TITLE
Check FSDP module has _device_mesh before getting it

### PR DESCRIPTION
**Pull request by eracah**
# What does this PR do?
If torch version <2.2.0 then FSDP module won't have `_device_mesh`, so `State.device_mesh` will raise an error. This PR does a check for `_device_mesh` first before getting the attribute

### CODE REVIEW

File: composer/core/state.py [hunk 1]
```diff
@@ -764,6 +764,8 @@ def fsdp_device_mesh(self):
         if self.fsdp_enabled:
             if not hasattr(self.model, 'model'):
                 return None
+            if not hasattr(self.model.model, '_device_mesh'):
+                return None
```
Comment (mvpatel2000) [inline composer/core/state.py:768]:
```suggestion
            if not hasattr(self.model, 'model') or not hasattr(self.model.model, '_device_mesh'):
                return None
```
Minor nit (I understand I gave the suggestion :P)

**Review by mvpatel2000 — APPROVED** [review]
LGTM!

File: composer/core/state.py (hunk 1 above)
Comment (eracah) [reply composer/core/state.py:768]: lol

Comment (eracah) [reply composer/core/state.py:768]: but then CI has to re-run

**Pull request merged**